components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java

```diff
@@ -1225,6 +1227,42 @@ private void publishEvent(User user, String notify, String code, String password
 
     }
 
+    private static void addUserStoreManagerToEventProperties(User user, HashMap<String, Object> properties) {
+
+        try {
+            int tenantId = IdentityTenantUtil.getTenantId(user.getTenantDomain());
+            RealmService realmService = IdentityRecoveryServiceDataHolder.getInstance().getRealmService();
+
+            if (realmService == null) {
+                if (log.isDebugEnabled()) {
+                    log.debug("RealmService is not available. Skipping setting user store manager.");
+                }
+                return;
+            }
+
+            UserRealm userRealm = realmService.getTenantUserRealm(tenantId);
+            if (userRealm == null) {
+                if (log.isDebugEnabled()) {
+                    log.debug("UserRealm is null for tenant: " + tenantId);
+                }
+                return;
+            }
+
+            UserStoreManager userStoreManager = userRealm.getUserStoreManager();
+            if (userStoreManager != null) {
+                properties.put(IdentityEventConstants.EventProperty.USER_STORE_MANAGER, userStoreManager);
+            } else if (log.isDebugEnabled()) {
+                log.debug("UserStoreManager is null for tenant: " + tenantId);
+            }
```

Comment:
Is this to get the user store manager id ? Given users user store manager is there can't we rely on that. Id is the base64encoded user store name right

Reply:
User manager is sent to retrieve userid and the email address of the user for the moment.